src/substitute.rs

```diff
@@ -65,18 +65,27 @@ fn substitute_line(line: &str, env: &Table) -> Result<String> {
     Ok(output)
 }
 
+// Only valid with ascii_alphabetic, ascii_digit or underscores in key name
+fn valid_character(c: &char) -> bool {
+    c.is_ascii_alphabetic() || c.is_ascii_digit() || *c == '_'
+}
+
 fn find_replacement(placeholder: &str, env: &Table) -> Result<String> {
     let mut parts = placeholder.split('|');
 
     let key = parts.next().unwrap_or("").trim();
-    match env.get(key) {
-        Some(Value::String(v)) => Ok(v.to_string()),
-        Some(Value::Integer(v)) => Ok(v.to_string()),
-        Some(Value::Float(v)) => Ok(v.to_string()),
-        Some(Value::Boolean(v)) => Ok(v.to_string()),
+    let parsed_key = key.chars().filter(valid_character).collect::<String>();
+
+    let parse = |v: &str| key.replace(&parsed_key, v);
```

Comment:
I'm totally confused. You'll need to explain what's going on here.

Reply:
I am collecting all `valid` characters of a name, to get the "real" name. Let's say I have this:

```
{
    "name": {{ "foo" | null }}
}
```

I will discard `"` and `parsed_key` will be `foo` and not `"foo"`. Then inside the `parse`-closure, I am replacing only `foo`. So if I input `bar` to inquire, or I have extracted something to the `foo`-variable, the result will end up being

```
{
    "name": "bar"
}
```

and not

```
{
    "name": bar
}
```

Reply:
It mostly means that I don't have to input the quotes to inquire